examples/distribution_to_json.py

```diff
@@ -0,0 +1,148 @@
+   '''
+
+from neurom import ezy
+from scipy import stats
+import numpy as np
+import argparse
+import json
+from collections import OrderedDict
+
+
+def parse_args():
+    '''Parse command line arguments'''
+    parser = argparse.ArgumentParser(
+        description='Morphology fit distribution extractor',
+        epilog='Note: Outputs json of the optimal distribution \
+                and corresponding parameters.')
+
+    parser.add_argument('datapath',
+                        help='Path to morphology data file or directory')
+
+    parser.add_argument('feature',
+                        help='Feature available for the ezy.neuron')
+
+    return parser.parse_args()
+
+
+def distribution_fit(data, distribution='norm'):
+    '''Calculates and returns the parameters of a distribution'''
+    return getattr(stats, distribution).fit(data)
+
+
+def distribution_error(data, distribution='norm'):
+    '''Calculates and returns the distance of a fitted distribution
+       from the initial data.
+    '''
+    params = distribution_fit(data, distribution=distribution)
+    return stats.kstest(data, distribution, params)[0]
+
+
+def test_multiple_distr(data):
+    '''Runs the distribution fit for multiple distributions and returns
+       the optimal distribution along with the corresponding parameters.
+    '''
+    distr_to_check = ['norm', 'expon', 'uniform']
+
+    fit_data = {d: distribution_fit(data, d) for d in distr_to_check}
+
+    fit_error = {distribution_error(data, d): d for d in distr_to_check}
+
+    optimal = fit_error.values()[np.argmax(fit_error.iterkeys())]
+
+    return optimal, fit_data[optimal]
+
+
+def extract_data(data_path, feature):
+    '''Loads a list of neurons, extracts feature
+       and transforms the fitted distribution in the correct format.
+       Returns the optimal distribution, corresponding parameters,
+       minimun and maximum values.
+    '''
+    population = ezy.load_neurons(data_path)
+
+    feature_data = [getattr(n, 'get_' + feature)() for n in population]
+
+    results = {}
+
+    try:
+        opt_fit = test_multiple_distr(feature_data)
+    except ValueError:
+        from itertools import chain
+        feature_data = list(chain(*feature_data))
+        opt_fit = test_multiple_distr(feature_data)
+
+    results["type"] = opt_fit[0]
+    results["params"] = opt_fit[1]
+    results["min"] = np.min(feature_data)
+    results["max"] = np.max(feature_data)
+
+    return results
+
+
+def transform_distribution(data):
+    '''Transform optimal distribution data into correct format
+       based on the distribution type.
+    '''
+    data_dict = OrderedDict()
+
+    if data["type"] == 'norm':
+        data_dict.update({"type": "normal"})
+        data_dict.update({"mu": data["params"][0]})
+        data_dict.update({"sigma": data["params"][1]})
+
+    elif data["type"] == 'expon':
+        data_dict.update({"type": "exponential"})
+        data_dict.update({"lambda": data["params"][1]})
```

Comment:
Ok, this needs to be hardcoded depending on the distribution. The problem is that each distribution fit in scipy returns a list of parameters. What each parameter means is unfortunately defined by the distribution you fit and there is no global way of defining this concept. Please don't merge this change before I make sure that for the exponential and normal I use the output of scipy in the right order. I will include a reference for each.